lib/validates_overlap/overlap_validator.rb

```diff
@@ -168,14 +168,28 @@ def value_attribute_name(attr_name)
     name + '_value'
   end
 
-  def resolve_scope_value(record, attr_name, value = nil)
+  def resolve_attribute_value(record, attr_name, value = nil)
     if value
       value.is_a?(Proc) ? value.call(record) : value
     else
-      record.read_attribute(attr_name)
+      value = record.read_attribute(attr_name)
+
+      if is_enum_attribute?(record, attr_name)
+        value = record.class.defined_enums[attr_name][value]
+      end
+
+      value
     end
   end
 
+  def is_enum_attribute?(record, attr_name)
+    implement_enum? && record.class.defined_enums[attr_name].present?
+  end
+
+  def implement_enum?
+    (ActiveRecord::VERSION::MAJOR > 5) || (ActiveRecord::VERSION::MAJOR > 4 && ActiveRecord::VERSION::MINOR > 1)
```

Comment:
@stricte thank you very much for finding the bug. I have fixed it. I know it took a long for such small thing, but I was quite busy. :) Anyway, thank you again :)